### DISCUSSION TITLE
How long does it take to train Tactron2_DDC ?

### DISCUSSION THREAD

**Discussion by koayst** [coqui-ai/TTS#1895] [General]
Hi,

I have been training a "recipes/ljspeech/tacotron2-DDC" model using 1 GPU of 16GB since Aug 15 and I noticed that I am at Epoch 83/1000.

At this speed, it is going to take months to hit 1000 epoch.  I didn't change any parameters found in train_tacotron2-DDC. Everything as cloned from github.

Am I doing it correctly ?  I have a 4 GPU system.  Should I switch to more GPU training ?

**Comment by thorstenMueller** [comment]
You do not have to train all 1000 epochs.

* Are you using GPU (CUDA) based training?
* Do you have tensorboard running?
* Are graphs looking good?
* At which step is your training process?

I've no experience with multi GPU training, but this might help improving training speed.

**Reply by koayst** [reply]
Hi @thorstenMueller,

- I am using GPU (with CUDA) to train.  
- I run tensorboard just now and it was at step 53,998.  I playback the audio and it sounded "robotic and broken".
- Estimate Epoch is around 133 (406 steps per epoch)
- Please see images below
<img width="557" alt="Capture02" src="https://user-images.githubusercontent.com/59504709/185932524-5a5315eb-6fda-4a57-8173-b6bfdf38ac6f.PNG">
<img width="453" alt="Capture01" src="https://user-images.githubusercontent.com/59504709/185932539-0498294a-9250-4fea-9560-2e40f1022d18.PNG">

I don't have much experience in this. Could you provide some guidance on what should I be looking for OR produce a video?  I learnt something when I watched your youtube channel.

**Reply by carrrlosss** [reply]
@koayst have you figured out the robotic and broken sound? i'm getting the same

**Reply by koayst** [reply]
No.  Didn't know what to do.  Hence, give up.

**Reply by carrrlosss** [reply]
i've completed 1000 epochs over 12hrs, still sounds like a robot

**Reply by siagupta-03** [reply]
> Hi @thorstenMueller,
> 
> * I am using GPU (with CUDA) to train.
> * I run tensorboard just now and it was at step 53,998.  I playback the audio and it sounded "robotic and broken".
> * Estimate Epoch is around 133 (406 steps per epoch)
> * Please see images below
> 
> <img alt="Capture02" width="557" src="https://user-images.githubusercontent.com/59504709/185932524-5a5315eb-6fda-4a57-8173-b6bfdf38ac6f.PNG">
> 
> <img alt="Capture01" width="453" src="https://user-images.githubusercontent.com/59504709/185932539-0498294a-9250-4fea-9560-2e40f1022d18.PNG">
> 
> I don't have much experience in this. Could you provide some guidance on what should I be looking for OR produce a video? I learnt something when I watched your youtube channel.

if not 1000 how much epochs are necessary

**Comment by thorstenMueller** [comment]
IMHO 50k steps is too low. Maybe check quality at around 100k.

**Comment by mahrukh667** [comment]
Training Setup:
Model: Tacotron 2
Dataset: ~10 hours of custom speech data
Batch Size: 32
 Epoch: 800
Learning Rate: 1e-5
Iteration: 34,906 (Epoch 167)
Issue:
The alignment graph (shown below) is not forming a clean diagonal, indicating poor attention learning between text (encoder) and speech (decoder).

Question:
What could be causing this non-diagonal alignment, and how can I fix it?
![image](https://github.com/user-attachments/assets/e02c534d-daf9-405d-b466-7a9bb79d2ecc)
